src/operations/operations-module.ts

```diff
@@ -100,7 +100,10 @@ export default class OperationsModule {
       this.logger.error(`Project ${projectId} not found in getMissingDeploymentActivity.`);
       throw Boom.badGateway();
     }
-    const mappedExisting = existing.map(item => ({
+    if (existing === null) {
+      throw Boom.badGateway();
+    }
+    const mappedExisting = (existing as any).map((item: any) => ({
```

Comment:
It seems that the reason you can't do `existing.map()` is that the `Promise.all` call above is incorrect. Shouldn't it be:
```ts
const [ expected, existing ] = await Promise.all([
  this.getProjectDeploymentActivity(projectId),
  this.activityModule.getProjectActivity(projectId),
]);
```
?